src/modules/bootloader/main.py

```diff
@@ -637,28 +637,33 @@ def run_grub_install(fw_type, partitions, efi_directory):
                                    "--bootloader-id=" + efi_bootloader_id,
                                    "--force"])
     else:
-        assert efi_directory is None
-        if libcalamares.globalstorage.value("bootLoader") is None:
-            return
+        if efi_directory is not None:
+            if not install_hybrid_grub:
+                return
 
-        boot_loader = libcalamares.globalstorage.value("bootLoader")
-        if boot_loader["installPath"] is None:
-            return
+        if libcalamares.globalstorage.value("bootLoader") is None:
+            find_esp_disk_command = "echo \"/dev/$(lsblk -o PKNAME $(df --output=source /boot/efi | tail -n1) | tail -n1)\""
+            boot_loader_install_path = check_target_env_output(["sh", "-c", find_esp_disk_command]).strip()
```

Comment:
If bootloader is None and fw_type is efi and install_hybrid_grub is false shouldn't we be not installing a bootloader here?

Reply:
Yes, but I think that's already the case, since the context is:

```python
if fw_type == "efi":
    ...do EFI things here...
else:
   ...
   libcalamares.globalstorage.value("bootLoader") is None:
        find_esp_disk_command = "long line of bash"
        boot_loader_install_path = check_target_env_output([...]).strip()
```

i.e., this code will only run if `fw_type` is NOT `efi`.